derrida/interventions/management/commands/insertion_data.py

```diff
@@ -0,0 +1,145 @@
+            self.base_filename = os.path.join(kwargs['directory'], self.base_filename)
+
+        # aggregate intervention data to be exported for use in generating
+        # CSV and JSON output
+
+        # canvas label indicates if a canvas is part of an insertion
+        insertion_canvases = Canvas.objects.filter(label__contains='Insertion')
+
+        insertion_images = defaultdict(list)
+        for canvas in insertion_canvases:
+            insertion_match = RE_INSERTION_LABEL.match(canvas.label)
+            if not insertion_match:
+                print('ERROR: regex fails on %s' % canvas.label)
+                continue
+
+            # construct group label from manifest label and insertion title within canvas label
+            # so we get unique image groups for each logical insertion
+            group_label = '%s. %s' % (canvas.manifest.label.strip('.'), insertion_match.group('label'))
+            insertion_images[group_label].append(canvas)
+
+        data = [self.insertion_data(label, canvas_group)
+                for label, canvas_group in insertion_images.items()]
+
+        # filter out any null values for canvases not linked to work instance
+        data = [d for d in data if d]
+
+        # list of dictionaries can be output as is for JSON export
+        with open('{}.json'.format(self.base_filename), 'w') as jsonfile:
+            json.dump(data, jsonfile, indent=2)
+
+        # generate CSV export
+        with open('{}.csv'.format(self.base_filename), 'w') as csvfile:
+            # write utf-8 byte order mark at the beginning of the file
+            csvfile.write(codecs.BOM_UTF8.decode())
+
+            csvwriter = csv.DictWriter(csvfile, fieldnames=self.csv_fields)
+            csvwriter.writeheader()
+
+            for insertion in data:
+                csvwriter.writerow(self.flatten_dict(insertion))
+
+    def insertion_data(self, label, canvases):
+        '''Generate a dictionary of data to export for a group
+        of canvases representing a single insertion in a single book.'''
+
+        # NOTE: using OrderedDict to ensure JSON output follows logical
+        # order in Python < 3.6, where dict order is not guaranteed
+
+        first_canvas = canvases[0]
+        # access book via manifest instance (reverse of digital edition relation)
+        # at least one insertion canvas is not related to a book...
+        try:
+            book = first_canvas.manifest.instance
+        except ObjectDoesNotExist:
+            print('manifest %s not related to work instance' % first_canvas.manifest.label)
+            # NOTE: one manifest is not linked, and it seems to be a duplicate;
+            # omit from export
+            book = None
+            return
+
+        # extract page number from first canvas label
+        page = RE_INSERTION_LABEL.match(first_canvas.label).group('page')
+
+        return OrderedDict([
+            ('id', label),   # provisional
+            ('intervention_type', 'insertion'),
```

Comment:
If it's called "insertion.json", I feel like this isn't necessary. Was there a reason you wanted to include this?

Reply:
I wanted to ask you about that. I was trying to keep the fields consistent with the annotations export, but there isn't all that much overlap. The two together are interventions; should we keep type implied by filename and rename the annotations export? I have a hard time imagining anyone looking at the two sets together (although I guess some possibilities for looking at overlap, heavy intervention within a volume...)

Reply:
Let me know what your decision may be!

Reply:
It seems clearer & simpler to rename the current intervention data to annotation data.

And then we can drop the type field, since it can be determined by the filename.